packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -0,0 +1,472 @@
+     */
+    name?: string;
+    /**
+     * The callback function to be invoked when the value of the input field changes
+     */
+    onChange?: OnEvent;
+    /**
+     * The callback function to be invoked when the the input field loses focus
+     *
+     * For React Native this will call `onEndEditing` event since we want to get the last value of the input field
+     */
+    onBlur?: OnEvent;
+    /**
+     * Used to turn the input field to controlled so user can control the value
+     */
+    value?: string;
+    /**
+     * Used to disable the input field
+     */
+    isDisabled?: boolean;
+    /**
+     * If true, the input is marked as required, and `required` attribute will be added
+     */
+    isRequired?: boolean;
+    /**
+     * Icon to be shown at the start of the input field
+     */
+    leadingIcon?: IconComponent;
+    /**
+     * Prefix symbol to be displayed at the beginning of the input field
+     */
+    prefix?: string;
+    /**
+     * this is left to the components which is extending BaseInput
+     *
+     * eg: consumers can render a loader or they could render a clear button
+     */
+    interactionElement?: ReactNode;
+    // /**
+    //  * Decides whether to render a clear icon button
+    //  */
+    // showClearButton?: boolean;
+    // /**
+    //  * Event handler to handle the onClick event for clear button.
+    //  */
+    // onClearButtonClick?: () => void;
+    // /**
+    //  * Decides whether to show a loading spinner for the input field.
+    //  */
+    // isLoading?: boolean;
+    /**
+     * Suffix symbol to be displayed at the beginning of the input field
+     */
+    suffix?: string;
+    /**
+     * Icon to be displayed at the end of the input field
+     */
+    trailingIcon?: IconComponent;
+    /**
+     * Displays the character counter under the input field
+     */
+    maxCharacters?: number;
+    /**
+     * alignment of the text inside input field
+     */
+    textAlign?: 'left' | 'center' | 'right';
+    /**
+     * If true, focuses the input field on load
+     *
+     * **Note:**
+     * Automatically focusing a form control can confuse visually-impaired people using screen-reading technology and people with cognitive impairments.
+     * When autofocus is assigned, screen-readers "teleport" their user to the form control without warning them beforehand.
+     */
+    autoFocus?: boolean;
+    /**
+     * determines what return key to show on the keyboard of mobile devices/virtual keyboard
+     * **Note**: Few values are platform dependent and might not render on all the platforms
+     *
+     * `enter` is only available on web
+     * `previous` is only available on native android
+     */
+    keyboardReturnKeyType?: 'enter' | 'go' | 'done' | 'next' | 'previous' | 'search' | 'send';
+    /**
+     * **Web only**
+     *
+     * Hints browser to display an appropriate virtual keyboard
+     *
+     *
+     */
+    inputMode?: 'text' | 'search' | 'tel' | 'email' | 'url';
```

Comment:
We accept `tel` here but `telephone` below for `autoCompleteSuggestionType`. Can we standardize this?